service/history/transferQueueStandbyTaskExecutor.go

```diff
@@ -294,11 +294,9 @@ func (t *transferQueueStandbyTaskExecutor) processCloseExecution(
 				Clock:                  executionInfo.ParentClock,
 			})
 			switch err.(type) {
-			case nil, *serviceerror.NotFound, *serviceerror.NamespaceNotFound, *serviceerror.Unimplemented:
-				// NOTE: NotFound is only returned when workflow already completed
-				// If workflow can't be found at all, WorkflowNotReady error will be returned.
+			case nil, *serviceerror.NamespaceNotFound, *serviceerror.Unimplemented:
 				return nil, nil
-			case *serviceerror.WorkflowNotReady:
+			case *serviceerror.NotFound, *serviceerror.WorkflowNotReady:
```

Comment:
Technically I can't do this. Will cause an issue when upgrading from 1.17.0 to this because when the new handler return not found, the old caller will ignore and skip the verification.

But the entire verification log is a new fix we added in 1.17, so maybe it's ok?

Reply:
This could only happen during a deployment?

Reply:
Yes.